### PR TITLE
feat(batch): optimize multi point get for RowSeqScan

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- There will be a bunch of pks needed to lookup for index lookup read pattern.
- ~~Avoid calling `execute_range` for this kind of multi point get read pattern.~~
- Try to merge all point get row into chunks.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Review by Little-Wallace — COMMENTED** [review]
Would you mind adding some test for point-get ?

**Review by BugenZhao — COMMENTED** [review]
What do you mean by the "overhead" of `execute_range`? Is it the number of chunk to yield?

**Comment by BugenZhao** [comment]
Selecting among a lot of streams (>1000) may also lead to too much polling overhead. 🥵 We may need to limit the concurrency. cc @xxchan

**Comment by chenzl25** [comment]
It seems multi point get produce too many single row chunks. Merging them into chunks is helpful.

**Comment by chenzl25** [comment]
> Would you mind adding some test for point-get ?

Let me try.

**Comment by Little-Wallace** [comment]
> What do you mean by the "overhead" of `execute_range`? Is it the number of chunk to yield?
> 
> It would be nice if we still make the `get_row` concurrent, while only returning a `Row` and constructing chunks with `DataChunkBuilder` outside. 🤔

I do not think so... 
A `get_row` operation has a lower probability of triggering IO. It is not necessary to make it concurrent....

**Comment by chenzl25** [comment]
> 

In my scenario, merging point get row into chunks work pretty well, so I think whether to make it run sequentially can leave it for later PR if it hurt performance in a real case.

File: src/batch/src/executor/row_seq_scan.rs [hunk 1]
```diff
@@ -303,18 +304,66 @@ impl<S: StateStore> RowSeqScanExecutor<S> {
             None
         };
 
-        // Scan all ranges concurrently.
-        let select_all = select_all(scan_ranges.into_iter().map(|scan_range| {
+        let (point_gets, range_scans): (Vec<ScanRange>, Vec<ScanRange>) = scan_ranges
```
Comment (xxchan) [inline src/batch/src/executor/row_seq_scan.rs:307]: Is it possible to have both point get and range scan at the same time? 🤔 If not, there's no need to partition.

Comment (chenzl25) [reply src/batch/src/executor/row_seq_scan.rs:307]: It could be. For example: `select * from t where id = 1 or id > 10`

**Pull request merged**